### ISSUE TITLE
Q: Are unquoted object keys in response examples intentional?

### ISSUE THREAD

**Issue by dcarr178** [mrin9/RapiDoc#245]
My users are asking why object keys are unquoted in response examples (non-valid json). When I click the copy button the text copied to clipboard contains quoted object keys, so I have to assume quotes were intentionally omitted from displaying but I'm not sure why. Perhaps for readability? I'm probably going to change this in my fork but can add a config option true/false users can pass into component if you think it's valuable enough to merge into master.

**Comment by mrin9** [comment]
There are various types of examples that can be provided. It can be JSON, XML, plain text, auto-generated etc.

all the examples are shown as it is provided in the spec, except in case of auto generated-json or valid JSON example, we use a tree like control to display it (it resembles a JSON structure but not a json)  this is done for readability and quick navigation (collapse/expand) purpose.
 
Even if you provide an quoted keys to make it  a valid json, it wont be valid the moment the user collapse one of the items, it will be shown something like below
```
{
name: "Mickey Mouse",
age: 10,
address: {. . .}
},
```
the other option would be to not use the tree-view al all, instead show it as a valid plain text value. 

The tree-view option for json examples was added based on user feedback,  we cannot alter the behavior unless it is backed up by separate user demand.

We do conduct monthly user feedback, I have added this suggestion to remove tree-view and show a valid json example using plain text. Will see what we get from that

**Comment by mrin9** [comment]
We just finished up our discussion on this, and majority(actually all) of our team members agrees with what you proposed. So we wont be waiting on the user feedback and instead go ahead and implement it 

The summary of our discussion,
> Showing a valid `JSON` in the example area is sematically correct thing to do, even if the user collapse a json node and make it invalid-json, it still falls in line with users expectation who is taking this action

So yes we would like to go with `quoted-keys` approach

Here is how we would like to address this 

- The examples are displayed using [`json-tree`](https://github.com/mrin9/RapiDoc/blob/master/src/components/json-tree.js) component
- Since this component is used internally in our **non-open-sourced** sections too, we like to support both options `quoted-keys` and `non-quoted-keys`
- So like to add an attribute `use-quoted-keys = "true"/"false"` to the  [`json-tree`](https://github.com/mrin9/RapiDoc/blob/master/src/components/json-tree.js#L12) component (must accept a string not boolean, if not provided assume the default value to be `"true"` )
- based on the attribute value it can render the key `quoted/un-quoted`

Let me know if you have any alternative thoughts on this and would like to make a PR. 

Thanks for bringing up this issue !!!

**Comment by mrin9** [comment]
closing due to inactivity

**Comment by joshrap-eb** [comment]
Would that `quoted-keys` approach ever be something you all would revisit? We did have a client bring up that the json was "missing" quotes. 

Not a big deal or anything since I just told them to hit the copy button instead of manually copying, but was still curious if this above approach would ever be added in a future release.

**Comment by seglo** [comment]
In my use case I'm trying to render an example that uses alphanumeric strings as object keys in a map. It would be nice to be able to enable quoting for examples like these.

i.e.

```json
{
	"01d0090f-0637-4d64-b688-ffc98695e4ad": "my value"
}
```
